external/mlir-hal/include/mlir/Dialect/MHAL/IR/MHALAttrDefs.td

```diff
@@ -79,4 +79,15 @@ def MHAL_KernelPackageAttr : MHAL_Attr<"KernelPackage"> {
     let hasCustomAssemblyFormat = 1;
 }
 
+// Misc Attrs
+def MHAL_PrefillAttr : MHAL_Attr<"Prefill"> {
+  let mnemonic = "mhal.prefill";
+
+  let parameters = (ins
+    "uint32_t":$argIndex,
+    "uint32_t":$initValue
+  );
+  let assemblyFormat = "`<` params `>`";
```

Comment:
Documentation?

Reply:
It is a good point!

I added something like

```mlir
  let description = [{
      A user interface parameter for initializing specific memory buffers

      The attribute is a way to describe which specific kernel parameter
      (typically memref) must be initialized with specific value before
      a kernel invocation.

  }];
```